workflows/smrnaseq.nf

```diff
@@ -124,26 +125,37 @@ workflow SMRNASEQ {
     // SUBWORKFLOW: Read QC and trim adapters
     //
 
+    if(!params.skip_qc){
     FASTQC_FASTP (
         ch_cat_fastq,
+        ch_fastp_adapters,
         false,
         false
     )
-
     ch_versions = ch_versions.mix(FASTQC_FASTP.out.versions)
-
     reads_for_mirna = FASTQC_FASTP.out.reads
 
+
     //
     // SUBWORKFLOW: mirtrace QC
     //
-    FASTQC_FASTP.out.adapterseq
-    | join( ch_cat_fastq )
-    | map { meta, adapterseq, fastq -> [meta + [adapter:adapterseq], fastq] }
-    | MIRTRACE
+    ch_outputs_for_mirtrace = FASTQC_FASTP.out.adapterseq
+    .join( ch_cat_fastq )
+    .map { meta, adapterseq, fastq -> [meta + [adapter:adapterseq], fastq] }
+    .collect()
+
+    MIRTRACE(ch_outputs_for_mirtrace)
 
     ch_versions = ch_versions.mix(MIRTRACE.out.versions.ifEmpty(null))
 
+    } else{
```

Comment:
This has been to be done still? otherwise remove

Reply:
This looks to me as though the workflow will fail if `params.skip_qc` is true. I'll add in catches to ensure that we don't end up with undefined channels to MultiQC.